db-functions

```diff
@@ -303,11 +303,7 @@ check_pkgfile() {
 
 	in_array "${pkgarch}" "${ARCHES[@]}" 'any' || return 1
 
-	if echo "${pkgfile##*/}" | grep -q "^${pkgname}-${pkgver}-${pkgarch}"; then
-		return 0
-	else
-		return 1
-	fi
+	[[ ${pkgfile##*/} == "${pkgname}-${pkgver}-${pkgarch}"* ]]
```

Comment:
Don't use ==, use = instead. This is a weird bash extension that literally serves no purpose as it merely aliases =.

Reply:
No, it is a cool bash extension that serves a purpose. You've added the only two examples of using == in the main code, just now, and I dispute Chet Ramey's reasons for adding the == alias to the bash scripting language.

Reply:
Glob/regex matching, but the rest of it is weird.  `[[ -r $filename ]]` adds nothing to the language, and serves no purpose over `[ -r "$filename" ]`

At the end of the day, it's just a style choice.  And that's fine.  But "it's the codebase's style" is the winning argument there, not "it's a bad extension".  I hadn't realized that the main codebase was all `=`; I'd though it was inconsistent; I hadn't realized that all of the `[[ == ]]` uses were in `test/`.